### PR TITLE
Add debugging to MTUSize test.

### PR DESCRIPTION
**Pull request by JamesWTruher**
Also be sure to include the reply in more conditions.
Fix up os version for Mac in get-platforminfo.

# PR Summary

## PR Context

The MTUSize test is somewhat flakey.
This hopes to improve that.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by JamesWTruher** [comment]
adding @vexx32 and @iSazonov  as reviewers as the reply will be returned more than previously. However, the test seems to be far more robust now

**Comment by JamesWTruher** [comment]
I'm explicitly ignoring the code factor issues

File: src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs [hunk 1]
```diff
@@ -585,6 +585,7 @@ private void ProcessMTUSize(string targetNameOrAddress)
                     if (reply.Status == IPStatus.PacketTooBig || reply.Status == IPStatus.TimedOut)
                     {
                         HighMTUSize = CurrentMTUSize;
+                        replyResult = reply;
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs:588]: looks like you now set `replyResult = reply` in each condition, so why not just have it set before the `if` statement?

**Review by SteveL-MSFT — DISMISSED** [review]

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs [hunk 2]
```diff
@@ -614,6 +615,7 @@ private void ProcessMTUSize(string targetNameOrAddress)
                         else
                         {
                             retry++;
+                            replyResult = reply;
```
[the remark this answers is not in the document]
Comment (JamesWTruher) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs:618]: @daxian-dbw - this is the reason that the test is flaky. Without returning the reply we cannot determine if the request has timed out. That's why I'm adding it under all these circumstances.

[the remark this answers is not in the document]
Comment (JamesWTruher) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs:618]: i suppose i could add the reply as the target object of the error, save the error in the test (via `-ErrorVariable`) and then check in that case

[the remark this answers is not in the document]
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs:618]: Sorry for the delay of getting back to this. Yes, saving the object in the error and checking the error in the test would be a better approach, as that won't change the intention of the method.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**